rumqttd/architecture.md

```diff
@@ -139,5 +139,94 @@ NOTE: Transition to ConnectionReady will schedule the connection for consumption
   - If ConnectionBusy
     - ConnectionBusy -> ConnectionReady
 
+## v2
+
+### Ontology
+
+Here are a few terms to understand before we move on. These terms are used internally, within the codebase, to organize the files and modules.
+
+#### Core Concepts and Entities
+
+0. Input and Output Queues (`ibufs` and `obufs`)
+
+   The basic foundation of `rumqttd` are two instances of `VecDequeue` from `std::collections`. Everything else just manages these and their interfaces to various other parts of the system. The queues are as follows -
+
+   - `ibufs`: All incoming data to `rumqttd` (represented by a `Packet`) is stored immediately in `ibufs` (which holds a `VecDequeue`)
+   - `obufs`: All outgoing data `rumqttd` (represented by `Notification`) is stored immediately in the `obufs` (which holds a `VecDequeue`).
+
+   `obufs` also has an _in-flight queue_ to track packets that have been sent.
+
+1. Broker
+
+   Top-level entity that handles _everything_, including, but not limited to
+
+   - Config Validation
+   - Configuring and creating the Router
+   - Starting the Router
+   - Maintaining a channel that communicates with the router
+
+   When you start the Broker, the following happens ->
+
+   1. The Router is started
```

Comment:
`Router` is spawned as soon as we create [new](https://github.com/bytebeamio/rumqtt/blob/714f71c9bc3369fa186353201dc2e8197d07594c/rumqttd/src/server/broker.rs#L68-L84) `Broker`, it happens before start.

Reply:
Changing this.